collect/central_collector.go

```diff
@@ -662,8 +870,10 @@ func (c *CentralCollector) makeDecisions(ctx context.Context) error {
 		if shouldSend {
 			state = centralstore.DecisionKeep
 			status.KeepReason = reason
+			c.Gossip.Publish(gossip_keep, []byte(trace.TraceID))
```

Comment:
Since we still set the status below, is there a possible race condition between the status in redis and the new `keepTraces`?

Reply:
Yes, which is why KeepTraces returns the list of things that didn't have a status available from Redis yet, so that it can be retried quickly.

Without it, it still works, but it has to wait for the cleanup task to find it, which might take several seconds. 

Noticing that delay is, in fact, why I added the retry logic.